packages/bot/src/scratch/utils.js

```diff
@@ -284,3 +284,16 @@ export const addDomAsBlock = (el_block, parent_block = null) => {
 
     return block;
 };
+
+export const saveWorkspace = () => {
+    const workspace_dom = Blockly.Xml.workspaceToDom(Blockly.derivWorkspace);
+    localStorage.setItem('workspace', Blockly.Xml.domToText(workspace_dom));
+};
+
+export const getPreviousWorkspace = () => {
+    if (localStorage.getItem('workspace')) {
```

Comment:
`localStorage.getItem` will return `null` if not set, so only return `localStorage.getItem('workspace')` will suffice.